### PR TITLE
Corrected zip key for IPStack

### PR DESCRIPTION
**Pull request by stevedrobinson**
| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [Y]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 

#### Description:
At some point IPStack changed the property/key used for zipcode from "zip_code" to "zip". This PR just corrects the key on the mautic side. 

#### Steps to test this PR:

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Get a free api key from ipstack.com
3. Update configuration to use IPStack.com as the IP Lookup Service and paste your api key into the IP lookup service authentication
image
4. In an incognito/private browsing window (and from a US IP address via proxy if necessary), visit the tracking pixel by replacing example.com with the hostname of the gitpod host: `https://example.com/mtracking.gif?page_url=test`
5. Search contacts for is:anonymous and review the location of the contact

<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/10722"></a>

### CODE REVIEW

**Comment by stevedrobinson** [comment]
I signed the contributor agreement earlier today (I thought I had done that before). Let me know if/when to resubmit.

**Review by escopecz — DISMISSED** [review]
This is correct change according to the official API docs: 

https://ipstack.com/documentation

File: app/bundles/CoreBundle/IpLookup/IpstackLookup.php [hunk 1]
```diff
@@ -30,7 +30,7 @@ protected function getUrl()
             $this->logger->warning('FreeGeoIP has become IPStack and now requires an API key.');
         }
 
-        return 'http://api.ipstack.com/'.$this->ip.'?access_key='.$this->auth.'&output=json&legacy=1';
+        return 'http://api.ipstack.com/' . $this->ip . '?access_key=' . $this->auth . '&output=json&legacy=1';
```
Comment (escopecz) [inline app/bundles/CoreBundle/IpLookup/IpstackLookup.php]: Please revert this change back as it breaks code style rules.
```suggestion
        return 'http://api.ipstack.com/'.$this->ip.'?access_key='.$this->auth.'&output=json&legacy=1';
```

**Comment by RCheesley** [comment]
@cla-bot check thanks for the PR @stevedrobinson :)

**Comment by stevedrobinson** [comment]
Fixed style issue.

**Comment by kuzmany** [comment]
> This branch is out-of-date with the base branch

@stevedrobinson can you rebase it onto 4.x?

**Comment by stevedrobinson** [comment]
@kuzmany done.

**Review by escopecz — DISMISSED** [review]
re-approving

**Review by kuzmany — DISMISSED** [review]

**Comment by kuzmany** [comment]
Need merge actual 4.x to this branch and also resolve unit tests. 
I cannot do it, because do not have permissions

**Comment by kuzmany** [comment]
Without touch, we cannot merged it to 4.2.
Add it to 4.2.1

**Comment by RCheesley** [comment]
@stevedrobinson can you please update the PR with the latest changes from the 4.2 branch? As this is on a fork under your org, maintainers are prevented from helping with this which slows progress considerably. If possible please try to make PRs on your own personal fork in the future!

**Comment by RCheesley** [comment]
@stevedrobinson we cannot consider this for the 4.2 release without your input - please can you review the requests above and action? We will be releasing on Monday 28th March.

**Comment by RCheesley** [comment]
Looks like this still has conflicts to resolve but unfortunately we can't help with that @stevedrobinson - are you able to get this resolved today? We have a release coming on Monday and would like to get this merged if we can.

**Comment by RCheesley** [comment]
Removing from the milestone due to unresolved conflicts. When this is fixed, please ping us and we can look to consider it for a future release!

**Comment by stevedrobinson** [comment]
@RCheesley @escopecz Let me know what you need from me to get this merged.

**Comment by escopecz** [comment]
@stevedrobinson thanks for sorting out the git issues. The last thing this needs is pass through the CI checks but I can see they are somehow stuck. I'll try to close and reopen this PR to see if that will kick them off.

**Comment by escopecz** [comment]
So  there is one more thing. This change broke this test:
```
1) Mautic\CoreBundle\Tests\Unit\IpLookup\IpstackLookupTest::testIpLookupSuccessful
Failed asserting that two strings are equal.
--- Expected
+++ Actual
@@ @@
-'94107'
+''

/home/runner/work/mautic/mautic/app/bundles/CoreBundle/Tests/Unit/IpLookup/IpstackLookupTest.php:32
```
Let me know if you need any assistance to get it fixed.

**Comment by stevedrobinson** [comment]
OK - I fixed the test so it should work now.

**Review by kuzmany — APPROVED** [review]

**Review by escopecz — APPROVED** [review]
Thanks!
